csm/core/controllers/information.py

```diff
@@ -77,3 +77,115 @@ async def post(self):
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query information about all resources from cortx topology."""
+        Log.debug("Handling GET request to query information about all resources from cortx topology.")
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        # Call Cortx Information Service
+        Log.info(f"Fetching cortx topology for resource:{resource}.")
+        response = await self._service.get_resources(resource)
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}/{resource_id}")
+class ResourceTopologyView(CsmView):
+    """
+    Get information about specific resource from cortx topology.
+    GET: Get information about specific resource from cortx topology.
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query information about specific resource from cortx topology."""
+        Log.debug("Handling GET request to query information about specific resource from cortx topology.")
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        resource_id = self.request.match_info[const.ARG_RESOURCE_ID]
+        Log.info(f"Fetching cortx topology for resource:{resource} and resource_id:{resource}.")
+        # Call Cortx Information Service
+        response = await self._service.get_specific_resource(resource, resource_id)
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}/{resource_id}/{view}")
+class AllViews(CsmView):
+    """
+    Get information about specific resource from cortx topology.
+    GET: Get information about specific resource from cortx topology.
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service =self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query information about all
+           views of specific resource from cortx topology."""
+        Log.debug("Handling GET request to query information about all views of specific resource from cortx topology.")
+        # Read path parameter
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        resource_id = self.request.match_info[const.ARG_RESOURCE_ID]
+        view = self.request.match_info[const.ARG_VIEW]
+        # Call Cortx Information Service
+        Log.info(f"Fetching cortx topology for resource:{resource}, resource_id:{resource} and view:{view}.")
+        response = await self._service.get_views(resource, resource_id, view)
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}/{resource_id}/{view}/{view_id}")
+class SpecificView(CsmView):
```

Comment:
Change the class name to something more appropiate

Reply:
Updated proper class name